src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2FilteringEngine.java

```diff
@@ -1,24 +1,50 @@
 package org.broadinstitute.hellbender.tools.walkers.mutect;
 
+import com.google.common.primitives.Doubles;
 import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFConstants;
+import org.apache.commons.math3.util.DoubleArray;
+import org.apache.commons.math3.util.MathArrays;
+import org.broadinstitute.hellbender.tools.walkers.contamination.ContaminationRecord;
 import org.broadinstitute.hellbender.utils.GATKProtectedVariantContextUtils;
 import org.broadinstitute.hellbender.utils.MathUtils;
 import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
 
 import java.util.*;
 
 /**
- * Created by davidben on 9/15/16.
+ * Created by David Benjamin on 9/15/16.
  */
 public class Mutect2FilteringEngine {
 
     public final static String ARTIFACT_IN_NORMAL_FILTER_NAME = "artifact_in_normal";
+    public final static String CONTAMINATION_FILTER_NAME = "contamination";
+
     public static final List<String> M_2_FILTER_NAMES = Arrays.asList(GATKVCFConstants.STR_CONTRACTION_FILTER_NAME, GATKVCFConstants.PON_FILTER_NAME,
             GATKVCFConstants.HOMOLOGOUS_MAPPING_EVENT_FILTER_NAME, GATKVCFConstants.CLUSTERED_EVENTS_FILTER_NAME,
             GATKVCFConstants.TUMOR_LOD_FILTER_NAME, GATKVCFConstants.GERMLINE_RISK_FILTER_NAME, GATKVCFConstants.TRIALLELIC_SITE_FILTER_NAME,
             GATKVCFConstants.STRAND_ARTIFACT_FILTER_NAME);
 
-    private Mutect2FilteringEngine() { }
+    private M2FiltersArgumentCollection MTFAC;
+    private final double contamination;
+
+
+    public Mutect2FilteringEngine(final M2FiltersArgumentCollection MTFAC) {
+        this.MTFAC = MTFAC;
+        contamination = MTFAC.contaminationTable == null ? 0.0 : ContaminationRecord.readContaminationTable(MTFAC.contaminationTable).get(0).getContamination();
+
+    }
+
+    // very naive M1-style contamination filter -- remove calls with AF less than the contamination fraction
+    private void applyContaminationFilter(final M2FiltersArgumentCollection MTFAC, final VariantContext vc, final Collection<String> filters) {
```

Comment:
Here we want to extract the tumor alt allele _fraction_ from the genotype field, not the population allele _frequency_ from the INFO field, right?